compliance/model/src/test/java/org/eclipse/rdf4j/model/util/IsomorphicBenchmark.java

```diff
@@ -0,0 +1,123 @@
+package org.eclipse.rdf4j.model.util;
+
+import org.eclipse.rdf4j.model.Model;
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.rio.Rio;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Benchmark)
+@Warmup(iterations = 20)
+@BenchmarkMode({ Mode.AverageTime })
+@Fork(value = 1, jvmArgs = { "-Xms1G", "-Xmx1G", "-XX:+UseSerialGC" })
+//@Fork(value = 1, jvmArgs = {"-Xms8G", "-Xmx8G", "-XX:+UseSerialGC", "-XX:+UnlockCommercialFeatures", "-XX:StartFlightRecording=delay=5s,duration=120s,filename=recording.jfr,settings=profile", "-XX:FlightRecorderOptions=samplethreads=true,stackdepth=1024", "-XX:+UnlockDiagnosticVMOptions", "-XX:+DebugNonSafepoints"})
+@Measurement(iterations = 10)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class IsomorphicBenchmark {
+
+	@Benchmark
+	public void empty() throws IOException {
+
+		isomorphic("empty.ttl");
+
+	}
+
+	@Benchmark
+	public void blankNodes() throws IOException {
+
+		isomorphic("blankNodes.ttl");
+
+	}
+
+	@Benchmark
+	public void shacl() throws IOException {
```

Comment:
Could you add a short javadoc comment that explains what each test actually does? E.g. "check performance on a typical SHACL shape with N blank nodes in a (chain/spread/tree/whatever) pattern". For some it's obvious but for others (e.g. the sparql test case) I'm not sure what you're actually testing.

Reply:
I can do that.

Reply:
this is still missing